### ISSUE TITLE
rules 'no-mixed-operators' and 'no-extra-parens' conflict

### ISSUE THREAD

**Issue by boggddan** [eslint/eslint#8891]
Hello, rules 'no-mixed-operators' and 'no-extra-parens' conflict

```javascript
/* eslint no-mixed-operators: "error"*/
/* eslint no-extra-parens: "error" */

// This code correct 'no-extra-parens'
var foo = ( a && b ) || c  ;

// This code correct for 'no-mixed-operators'
var foo = a && b || c ;
```

output from command line
```bash
eslint src/js/test4.js

D:\ProjectRuby\table_test\src\js\test4.js
  5:11  error  Gratuitous parentheses around expression  no-extra-parens
  8:13  error  Unexpected mix of '&&' and '||'           no-mixed-operators
  8:18  error  Unexpected mix of '&&' and '||'           no-mixed-operators
```

.eslintrc.js
```javascript
module.exports = {
  env: {
    browser: true,
    es6: true,
    jquery: true,
    node: false
  },
  extends: '',
  parserOptions: {
    ecmaVersion: 2017,
    sourceType: 'module',
    ecmaFeatures: {
      experimentalObjectRestSpread: true
    },
  },
  rules: {
  }
};
```

**Comment by not-an-aardvark** [comment]
Thanks for reporting. This is a known conflict -- you can use the `nestedBinaryExpressions: false` option in `no-extra-parens` to get around it.

**Comment by not-an-aardvark** [comment]
Closing because this is working as intended.
